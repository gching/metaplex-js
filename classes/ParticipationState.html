<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>ParticipationState | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="ParticipationState.html">ParticipationState</a>
				</li>
			</ul>
			<h1>Class ParticipationState</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">ParticipationState</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="ParticipationState.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="ParticipationState.html#collectedToAcceptPayment" class="tsd-kind-icon">collected<wbr>ToAccept<wbr>Payment</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="ParticipationState.html#primarySaleHappened" class="tsd-kind-icon">primary<wbr>Sale<wbr>Happened</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="ParticipationState.html#printingAuthorizationTokenAccount" class="tsd-kind-icon">printing<wbr>Authorization<wbr>Token<wbr>Account</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="ParticipationState.html#validated" class="tsd-kind-icon">validated</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="ParticipationState.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="ParticipationState.html#isPrimarySaleHappened" class="tsd-kind-icon">is<wbr>Primary<wbr>Sale<wbr>Happened</a></li>
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="ParticipationState.html#isValidated" class="tsd-kind-icon">is<wbr>Validated</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Participation<wbr>State<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-symbol">{ </span>collectedToAcceptPayment<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">BN</span><span class="tsd-signature-symbol">; </span>primarySaleHappened<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol">; </span>validated<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> }</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="ParticipationState.html" class="tsd-signature-type" data-tsd-kind="Class">ParticipationState</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L37">models/programs/metaplex/ParticipationState.ts:37</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5><span class="tsd-flag ts-flagOptional">Optional</span> args: <span class="tsd-signature-symbol">{ </span>collectedToAcceptPayment<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">BN</span><span class="tsd-signature-symbol">; </span>primarySaleHappened<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol">; </span>validated<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> }</span></h5>
									<ul class="tsd-parameters">
										<li class="tsd-parameter">
											<h5><span class="tsd-flag ts-flagOptional">Optional</span> collected<wbr>ToAccept<wbr>Payment<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">BN</span></h5>
										</li>
										<li class="tsd-parameter">
											<h5><span class="tsd-flag ts-flagOptional">Optional</span> primary<wbr>Sale<wbr>Happened<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></h5>
										</li>
										<li class="tsd-parameter">
											<h5><span class="tsd-flag ts-flagOptional">Optional</span> validated<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></h5>
										</li>
									</ul>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="ParticipationState.html" class="tsd-signature-type" data-tsd-kind="Class">ParticipationState</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="collectedToAcceptPayment" class="tsd-anchor"></a>
					<h3>collected<wbr>ToAccept<wbr>Payment</h3>
					<div class="tsd-signature tsd-kind-icon">collected<wbr>ToAccept<wbr>Payment<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BN</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L24">models/programs/metaplex/ParticipationState.ts:24</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="primarySaleHappened" class="tsd-anchor"></a>
					<h3>primary<wbr>Sale<wbr>Happened</h3>
					<div class="tsd-signature tsd-kind-icon">primary<wbr>Sale<wbr>Happened<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> = 0</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L25">models/programs/metaplex/ParticipationState.ts:25</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="printingAuthorizationTokenAccount" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> printing<wbr>Authorization<wbr>Token<wbr>Account</h3>
					<div class="tsd-signature tsd-kind-icon">printing<wbr>Authorization<wbr>Token<wbr>Account<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L27">models/programs/metaplex/ParticipationState.ts:27</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="validated" class="tsd-anchor"></a>
					<h3>validated</h3>
					<div class="tsd-signature tsd-kind-icon">validated<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span><span class="tsd-signature-symbol"> = 0</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L26">models/programs/metaplex/ParticipationState.ts:26</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L6">models/programs/metaplex/ParticipationState.ts:6</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isPrimarySaleHappened" class="tsd-anchor"></a>
					<h3>is<wbr>Primary<wbr>Sale<wbr>Happened</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isPrimarySaleHappened<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L29">models/programs/metaplex/ParticipationState.ts:29</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isValidated" class="tsd-anchor"></a>
					<h3>is<wbr>Validated</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isValidated<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/5d5013b/src/models/programs/metaplex/ParticipationState.ts#L33">models/programs/metaplex/ParticipationState.ts:33</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="ParticipationState.html" class="tsd-kind-icon">Participation<wbr>State</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="ParticipationState.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="ParticipationState.html#collectedToAcceptPayment" class="tsd-kind-icon">collected<wbr>ToAccept<wbr>Payment</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="ParticipationState.html#primarySaleHappened" class="tsd-kind-icon">primary<wbr>Sale<wbr>Happened</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="ParticipationState.html#printingAuthorizationTokenAccount" class="tsd-kind-icon">printing<wbr>Authorization<wbr>Token<wbr>Account</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="ParticipationState.html#validated" class="tsd-kind-icon">validated</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="ParticipationState.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="ParticipationState.html#isPrimarySaleHappened" class="tsd-kind-icon">is<wbr>Primary<wbr>Sale<wbr>Happened</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="ParticipationState.html#isValidated" class="tsd-kind-icon">is<wbr>Validated</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>