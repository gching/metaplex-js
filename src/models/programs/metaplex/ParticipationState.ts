import BN from 'bn.js';
import { PublicKey } from '@solana/web3.js';
import { BorshSchema } from '../../utils/types';

export class ParticipationState {
  static SCHEMA: BorshSchema = new Map([
    [
      ParticipationState,
      {
        kind: 'struct',
        fields: [
          ['collectedToAcceptPayment', 'u64'],
          ['primarySaleHappened', 'u8'], //bool
          ['validated', 'u8'], //bool
          [
            'printingAuthorizationTokenAccount',
            { kind: 'option', type: 'pubkey' },
          ],
        ],
      },
    ],
  ]);

  collectedToAcceptPayment: BN = new BN(0);
  primarySaleHappened: 0 | 1 = 0;
  validated: 0 | 1 = 0;
  printingAuthorizationTokenAccount?: PublicKey = undefined;

  get isPrimarySaleHappened(): boolean {
    return this.primarySaleHappened === 1;
  }

  get isValidated(): boolean {
    return this.validated === 1;
  }

  constructor(args?: {
    validated?: 0 | 1;
    collectedToAcceptPayment?: BN;
    primarySaleHappened?: 0 | 1;
  }) {
    Object.assign(this, args);
  }
}
