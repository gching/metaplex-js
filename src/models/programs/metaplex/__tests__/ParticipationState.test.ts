import { ParticipationState } from '../ParticipationState';
import BN from 'bn.js';

describe('ParticipationState', () => {
  it('should instantiate with the correct values', () => {
    const inst = new ParticipationState({
      primarySaleHappened: 0,
      validated: 1,
      collectedToAcceptPayment: new BN(0),
    });

    expect(inst).toMatchSnapshot();
  });
});
