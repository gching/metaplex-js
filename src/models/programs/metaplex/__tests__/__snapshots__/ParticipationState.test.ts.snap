// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipationState should instantiate with the correct values 1`] = `
ParticipationState {
  "collectedToAcceptPayment": "00",
  "primarySaleHappened": 0,
  "printingAuthorizationTokenAccount": undefined,
  "validated": 1,
}
`;
